File: 3396-valid-word/valid-word.cpp
class Solution {
public:
    bool isValid(string word) {
        if (word.length() < 3) return false;

        bool hasVowel = false;
        bool hasConsonant = false;
        string vowels = "aeiouAEIOU";

        for (char ch : word) {
            if (!isalnum(ch)) return false; // Not a letter or digit

            if (isalpha(ch)) {
                if (vowels.find(ch) != string::npos) {
                    hasVowel = true;
                } else {
                    hasConsonant = true;
                }
            }
        }

        return hasVowel && hasConsonant;
    }
};

File: 3396-valid-word/valid-word.swift
class Solution {
    func isValid(_ word: String) -> Bool {
        if word.count < 3 {
            return false
        }

        let vowels: Set<Character> = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
        var hasVowel = false
        var hasConsonant = false

        for ch in word {
            if !ch.isLetter && !ch.isNumber {
                return false
            }

            if ch.isLetter {
                if vowels.contains(ch) {
                    hasVowel = true
                } else {
                    hasConsonant = true
                }
            }
        }

        return hasVowel && hasConsonant
    }
}
